### ISSUE TITLE
For each line in Snippet

### ISSUE THREAD

**Issue by blueray453** [microsoft/vscode#92987]
I have lines like:
```
first
second
third
```
I want to convert them to (using snippet, NOT multi-cursor):

```
. first
. second
. third
```
I tried the following in custom snippets:
```
"Ordered List": {
	"scope": "asciidoc,plaintext",
	"prefix": "ol",
	"body": [
		". $TM_CURRENT_LINE"
	],
	"description": "Ordered List"
}
```
This converts

```
first
second
third
```

to 

```
. first
```

I want something like **for each lines** instead of `$TM_CURRENT_LINE`. How can I achieve that?

**Comment by jrieken** [comment]
Like so 👇 

![Uploading Mar-19-2020 16-50-45.gif…]()

**Comment by blueray453** [comment]
@jrieken yes, this is what exactly I am looking for. Can you please share the snippet with us.
.

**Comment by jrieken** [comment]
This is the snippet you have posted above

**Comment by blueray453** [comment]
I got another solution using `yhirose.FilterText` plugins. 

keybindings.json:
```
{
        "key": "ctrl+shift+,",
        "command": "extension.filterPredefined",
        "when": "editorFocus && editorLangId =~ /asciidoc|plaintext/"
},
```
settings.json:
```
"filterText.commandList": [
        {
            "name": "ol",
            "description": "Adoc OL",
            "command": "awk '$0=\". \"$0'"
        }
    ],
```
I think https://code.visualstudio.com/docs/editor/userdefinedsnippets#_grammar needs to include foreach line. I tried the process shown above but it is just too much work. 

It will also be great to have an option to process selected text with scripts (bash, python etc.), like my shown solution, **using snippets**.  That way we don't have to remember whether to invoke snippets or FilterText.

I know the second proposal might affects the portability of the snippet (windows/linux, has Python installed or not) but I think this idea is worth discussion at least.
